src/functions/getAddresses.ts

```diff
@@ -0,0 +1,231 @@
+import bitwise from 'bitwise';
+import { Byte, UInt4 } from 'bitwise/types';
+import {
+  ADDR_STR_LEN,
+  decResLengths,
+  EXTERNAL,
+  getFwVersionConst,
+} from '../constants';
+import { decryptResponse, encryptRequest, request } from '../shared/functions';
+import {
+  validateFwVersion,
+  validateIsUInt4,
+  validateNAddresses,
+  validateSharedSecret,
+  validateStartPath,
+  validateUrl,
+  validateWallet,
+} from '../shared/validators';
+import { isValidAssetPath } from '../util';
+
+/**
+ * `getAddresses` takes a starting path and a number to get the addresses associated with the
+ * active wallet.
+ * @category Lattice
+ * @returns An array of addresses.
+ */
+export async function getAddresses ({
+  startPath,
+  n,
+  flag: _flag,
```

Comment:
I'm really confused by this syntax. What is `_flag` here and how is it different than `flag`? And I see you redeclare `flag` as a const below. Double confused. Maybe I've been reviewing for too long...

Reply:
This syntax was all over the place so I just ignored the other instances but yeah I am still confused about how it works.

Reply:
Ohhh yeahh--basically, I wanted to be able to use the variable name, "flag" farther down the function instead of something like "validFlag" or "flagTwo" or something. So I alias `flag` to `_flag` in the params, which lets me refer to that argument as `_flag` in the function. 

`validateGetAddressesRequest` then accepts an object with the key of `flag` but I only have `_flag`, so I pass it `{flag: _flag}`, and then I get out a nice clean `flag` const that I can use later in the function. 

But I've just realized that I don't really need to get a `flag` value out of the validator function, so it's all moot anyway.